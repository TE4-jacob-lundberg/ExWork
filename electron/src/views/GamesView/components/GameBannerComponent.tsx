import React, { useState } from 'react';
import Styled from '@emotion/styled';

import { IGame } from '../../../shared/constants/standardGames';

interface Props {
  game: IGame;
}

const GameBannerComponent: React.FC<Props> = function (props: Props) {
  const [grayScale, setGrayScale] = useState('50');

  function handleClick(): void {
    // Implement click functionality
  }

  function handleMouseEnter(): void {
    setGrayScale('0');
  }

  function handleMouseLeave(): void {
    setGrayScale('50');
  }

  const ContainerStyled = Styled.div`
    height: 100%;
    position: relative;
    overflow: hidden;
    display: flex;
    justify-content: center;
  `;

  const BackgroundStyled = Styled.img`
    position: absolute;
    filter: grayscale(${grayScale}%);
    height: 100%;
    z-index: -1;
  `;

  return (
    <ContainerStyled onMouseEnter={handleMouseEnter} onMouseLeave={handleMouseLeave} onClick={handleClick} >
      <BackgroundStyled src={require(`../../../shared/assets/images/${props.game.image.name}`)} />
    </ContainerStyled>
  );
};

export default GameBannerComponent;
