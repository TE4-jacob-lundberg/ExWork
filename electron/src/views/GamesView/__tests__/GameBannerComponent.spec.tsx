import React from 'react';
import { RenderResult } from '@testing-library/react';
import renderer from 'react-test-renderer';

import GameBannerComponent from '../components/GameBannerComponent';

function factory(): RenderResult {
  return renderer.create(
    <GameBannerComponent
      game={{
        title: 'Game 1',
        abbreviation: 'G1',
        image: {
          name: 'csgo.jpg',
        },
      }}       
    />
  );
}

describe('GameBannerComponent.tsx', () => {
  it('renders correctly', () => {
    const container = factory();
    
    expect(container).toMatchSnapshot();
  });
});
