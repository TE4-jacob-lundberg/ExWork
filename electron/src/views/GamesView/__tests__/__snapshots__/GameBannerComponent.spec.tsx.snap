// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GameBannerComponent.tsx redirects to its links on click 1`] = `
<div>
  <div
    class="css-1d2zccn"
    data-testid="banner"
  >
    <div
      class="css-1uckm9l"
    >
      <button
        class="css-1lidn5m"
      >
        <i
          class="material-icons"
        >
          edit
        </i>
      </button>
    </div>
  </div>
</div>
`;

exports[`GameBannerComponent.tsx renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="css-1d2zccn"
        data-testid="banner"
      >
        <div
          class="css-1uckm9l"
        >
          <button
            class="css-1lidn5m"
          >
            <i
              class="material-icons"
            >
              edit
            </i>
          </button>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="css-1d2zccn"
      data-testid="banner"
    >
      <div
        class="css-1uckm9l"
      >
        <button
          class="css-1lidn5m"
        >
          <i
            class="material-icons"
          >
            edit
          </i>
        </button>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
